lib/libutee/tee_api_operations.c

```diff
@@ -860,75 +860,141 @@ TEE_Result TEE_CipherDoFinal(TEE_OperationHandle op,
 
 /* Cryptographic Operations API - MAC Functions */
 
-void TEE_MACInit(TEE_OperationHandle operation, const void *IV, uint32_t IVLen)
+void TEE_MACInit(TEE_OperationHandle operation, void *IV, uint32_t IVLen)
 {
 	TEE_Result res;
+	void *iv;
+	uint32_t ivlen;
 
 	if (operation == TEE_HANDLE_NULL)
 		TEE_Panic(0);
-	if (IV == NULL && IVLen != 0)
+	if (!operation->key1)
 		TEE_Panic(0);
 	if (operation->info.operationClass != TEE_OPERATION_MAC)
 		TEE_Panic(0);
-	res = utee_hash_init(operation->state, IV, IVLen);
+
+	switch (operation->info.algorithm) {
+	case TEE_ALG_DES_CBC_MAC_NOPAD:
+	case TEE_ALG_AES_CBC_MAC_NOPAD:
+	case TEE_ALG_AES_CBC_MAC_PKCS5:
+	case TEE_ALG_AES_CMAC:
+	case TEE_ALG_DES_CBC_MAC_PKCS5:
+	case TEE_ALG_DES3_CBC_MAC_NOPAD:
+	case TEE_ALG_DES3_CBC_MAC_PKCS5:
+		if (IV == NULL && IVLen != 0)
+			TEE_Panic(0);
+		iv = IV;
+		ivlen = IVLen;
+		break;
+
+	default:
+		iv = NULL;
```

Comment:
Even if currently not implemented, algos that could use IV should have coherent parameters. Do you agree ?